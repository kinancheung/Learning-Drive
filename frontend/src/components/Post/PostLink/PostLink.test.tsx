import { render } from '@testing-library/react';

import { PostLink } from '.';

describe('PostLink', () => {
  it('renders without error', () => {
    const view = render(<PostLink url='https://www.google.com' />);

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders title correctly', () => {
    const view = render(<PostLink url='https://www.google.com' title='Google' />);

    expect(view.asFragment()).toMatchSnapshot();
  });
});
