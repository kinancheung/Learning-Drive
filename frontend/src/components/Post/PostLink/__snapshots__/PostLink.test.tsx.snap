// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostLink renders title correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1ydk5vu-MuiPaper-root-MuiCard-root"
  >
    <a
      class="MuiButtonBase-root MuiCardActionArea-root css-1utj8hy-MuiButtonBase-root-MuiCardActionArea-root"
      href="https://www.google.com"
      rel="noopener noreferrer"
      tabindex="0"
      target="_blank"
    >
      <div
        class="MuiBox-root css-puscwe"
      />
      <div
        class="MuiCardContent-root css-qedw9d-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          https://www.google.com
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1jflqzp-MuiTypography-root"
        >
          Google
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;

exports[`PostLink renders without error 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1ydk5vu-MuiPaper-root-MuiCard-root"
  >
    <a
      class="MuiButtonBase-root MuiCardActionArea-root css-1utj8hy-MuiButtonBase-root-MuiCardActionArea-root"
      href="https://www.google.com"
      rel="noopener noreferrer"
      tabindex="0"
      target="_blank"
    >
      <div
        class="MuiBox-root css-puscwe"
      />
      <div
        class="MuiCardContent-root css-qedw9d-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          https://www.google.com
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1jflqzp-MuiTypography-root"
        >
          https://www.google.com
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
